### PR TITLE
[SPARK-8657] [YARN] Simplify method addResource() and get the correct fs from destPath.

### PR DESCRIPTION
**Pull request by litao-buptsse**
[SPARK-8657] [YARN] Simplify method addResource() and get the correct fs from destPath.

JIRA Link: https://issues.apache.org/jira/browse/SPARK-8657

### CODE REVIEW

**Comment by srowen** [comment]
OK to test

**Comment by litao-buptsse** [comment]
Some unit tests failed. I will review my PR code again.

**Comment by litao-buptsse** [comment]
@srowen I think maybe the first fs  param in addResource() is needed, especially it's useful when doing unit test. I see the ClientDistributedCacheManagerSuite usually pass a mock fs to the addResource(). So I think to keep the first fs  param might be better, although it's sometimes redundant.

ClientDistributedCacheManagerSuite unit tests: https://github.com/apache/spark/blob/master/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientDistributedCacheManagerSuite.scala

My failed unit tests: https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/967/testReport/org.apache.spark.deploy.yarn/

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -322,7 +322,7 @@ private[spark] class Client(
             destName.orElse(Option(localURI.getFragment())).getOrElse(localPath.getName())
           val destPath = copyFileToRemote(dst, localPath, replication)
           distCacheMgr.addResource(
-            fs, hadoopConf, destPath, localResources, resType, linkname, statCache,
```
Comment (lianhuiwang) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: i think we can replace fs with destPath's FileSystem. like: 
val fs = FileSystem.get(destPath.toUri(), hadoopConf)

**Comment by srowen** [comment]
OK to test

**Pull request closed without merging**